src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs

```diff
@@ -1909,7 +1909,7 @@ public enum GCS : uint
                     var result = ImmGetCompositionString(hIMC, dwIndex, (IntPtr)bufferPtr, (uint)bufferLength);
                     if (result >= 0)
                     {
-                        return Marshal.PtrToStringUni((IntPtr)bufferPtr);
+                        return Encoding.Unicode.GetString(buffer.ToArray());
```

Comment:
We use stackalloc specifically to avoid array allocation. Consider using the overload with pointers instead

Reply:
You are right! Hope I can do the right thing this time...